framework/src/wifi_manager/wifi_manager.c

```diff
@@ -337,37 +441,48 @@ wifi_manager_result_e wifi_manager_deinit()
 		return WIFI_MANAGER_FAIL;
 	}
 
+	g_manager_info.mode = WIFI_NONE;
+
 	return WIFI_MANAGER_SUCCESS;
 }
 
 wifi_manager_result_e wifi_manager_set_mode(wifi_manager_mode_e mode, wifi_manager_softap_config_s *config)
 {
 	wifi_utils_result_e result = WIFI_UTILS_SUCCESS;
 
-	if (mode != STA_MODE && mode != SOFT_AP_MODE) {
+	if (mode != STA_MODE && mode != SOFTAP_MODE) {
+		ndbg("wifi manager set mode failed: invalid mode\n");
 		return WIFI_MANAGER_INVALID_ARGS;
 	}
-	// STA mode -> SOFT AP mode
-	if (g_manager_info.mode == STA_MODE && mode == SOFT_AP_MODE) {
-		wifi_utils_softap_config_s soft_ap_config;
 
-		soft_ap_config.channel = config->channel;
-		soft_ap_config.ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
-		soft_ap_config.ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
-		strncpy(soft_ap_config.ssid, config->ssid, sizeof(config->ssid));
-		soft_ap_config.ssid_length = strlen(config->ssid);
-		strncpy(soft_ap_config.passphrase, config->passphrase, sizeof(config->passphrase));
-		soft_ap_config.passphrase_length = strlen(config->passphrase);
-		soft_ap_config.inform_new_sta_join = g_manager_info.wmcb->softap_sta_join;
+	if (g_manager_info.mode == mode) {
+		ndbg("wifi manager set mode failed: current mode is the same as requested.\n");
+		return WIFI_MANAGER_SUCCESS;
+	}
+
+	ndbg("required config: %d %d %s\n", mode, config->channel, config->ssid);
```

Comment:
We don't need to check whether config is NULL or not?
If config is NULL with configuration for ndbg, NULL dereferencing will happen I think.

Reply:
I saw that config is not used in STA_MODE mode now.
I think you'd better to move this comment below or add condition about checking NULL.

Reply:
I remove it in #554.